ros2_batch_job/__main__.py

```diff
@@ -401,6 +366,19 @@ def build_and_test(args, job):
         test_cmd.append('--pytest-with-coverage')
     test_cmd.extend(args.test_args)
 
+    # In Foxy and prior, xunit2 format is needed to make Jenkins xunit plugin 2.x happy
+    # After Foxy, we introduced per-package changes to make local builds and CI
+    # builds act the same.
+    if args.ros_distro in ('dashing', 'eloquent', 'foxy'):
+        pytest_args = ['-o', 'junit_family=xunit2']
+        # We should only have one --pytest-args option, or some options might get ignored
+        if '--pytest-args' in test_cmd:
+            pytest_opts_index = test_cmd.index('--pytest-args') + 1
+            test_cmd = test_cmd[:pytest_opts_index] + pytest_args + test_cmd[pytest_opts_index:]
```

Comment:
Suggestion:
```suggestion
            test_cmd.insert(pytest_opts_index, pytest_args)
```

Reply:
This will insert `pytest_args` list as a single element, instead of keeping `test_cmd` a single list of strings. If you think it's more clear or succinct, we can do:

```python
for i, opt in enumerate(pytest_args):
    test_cmd.insert(pytest_opts_index + i, opt)
```

I don't have a preference.

Reply:
Oh, you're totally right. No, what you have is great.